mindsdb/utilities/ml_task_queue/consumer.py

```diff
@@ -0,0 +1,253 @@
+            self.consumer_group.streams[TASKS_STREAM_NAME].ack(message_id)
+
+            payload = pickle.loads(message_content[b'payload'])
+            task_type = ML_TASK_TYPE(message_content[b'task_type'])
+            model_id = int(message_content[b'model_id'])
+            company_id = message_content[b'company_id']
+            if len(company_id) == 0:
+                company_id = None
+            redis_key = RedisKey(message_content.get(b'redis_key'))
+
+            # region read dataframe
+            dataframe_bytes = self.cache.get(redis_key.dataframe)
+            dataframe = None
+            if dataframe_bytes is not None:
+                dataframe = from_bytes(dataframe_bytes)
+                self.cache.delete(redis_key.dataframe)
+            # endregion
+
+            ctx.load(payload['context'])
+        finally:
+            self._ready_event.set()
+
+        try:
+            task = process_cache.apply_async(
+                task_type=task_type,
+                model_id=model_id,
+                payload=payload,
+                dataframe=dataframe
+            )
+            status_notifier = StatusNotifier(redis_key, ML_TASK_STATUS.PROCESSING, self.db, self.cache)
+            status_notifier.start()
+            result = task.result()
+        except Exception as e:
+            status_notifier.stop()
+            exception_bytes = to_bytes(e)
+            self.cache.set(redis_key.exception, exception_bytes, 10)
+            self.db.publish(redis_key.status, ML_TASK_STATUS.ERROR.value)
+            self.cache.set(redis_key.status, ML_TASK_STATUS.ERROR.value, 180)
+        else:
+            status_notifier.stop()
+            if isinstance(result, DataFrame):
+                dataframe_bytes = to_bytes(result)
+                self.cache.set(redis_key.dataframe, dataframe_bytes, 10)
+            self.db.publish(redis_key.status, ML_TASK_STATUS.COMPLETE.value)
+            self.cache.set(redis_key.status, ML_TASK_STATUS.COMPLETE.value, 180)
+
+    def run(self) -> None:
+        """ Start new listen thread each time when _ready_event is set
+        """
+        self._ready_event.set()
+        while self._stop_event.is_set() is False:
+            self._ready_event.wait(timeout=1)
+            if self._ready_event.is_set() is False:
+                continue
+            self._ready_event.clear()
+            threading.Thread(target=self._listen).start()
+
+    def stop(self) -> None:
+        """ Stop all executing threads
+        """
+        self._stop_event.set()
+        for thread in (*self._listen_message_threads, self._collect_cpu_stat_thread):
+            try:
+                if thread.is_alive():
+                    thread.join()
+            except Exception:
+                pass
+
+
+@mark_process(name='internal', custom_mark='ml_task_consumer')
+def start(verbose: bool) -> None:
+    """ Create task queue consumer and start listen the queue
+    """
+    try:
+        consumer = MLTaskConsumer()
```

Comment:
if consumer variable is used in except block - maybe put this line outside of try block